pkg/generate/app/imagestreamlookup.go

```diff
@@ -109,7 +109,7 @@ func InputImageFromMatch(match *ComponentMatch) (*ImageRef, error) {
 		if err != nil {
 			return nil, err
 		}
-		input.AsImageStream = true
+		input.AsImageStream = !match.LocalOnly
```

Comment:
For this case because you matched an image stream, LocalOnly would never be true, so I think it's fine to always set AsImageStream to true.